pandas/core/reshape.py

```diff
@@ -202,10 +202,13 @@ def get_new_columns(self):
                 return self.removed_level
 
             lev = self.removed_level
-            vals = np.insert(lev.astype('object'), 0,
+            vals = np.insert(lev.values.astype('object'), 0,
                              _get_na_value(lev.dtype.type))
 
-            return lev._shallow_copy(vals)
+            try:
+                return lev._shallow_copy(vals.astype(lev.dtype, copy=False))
+            except ValueError:
```

Comment:
few lines above it is inserting `_get_na_value(lev.dtype.type)` into the `lev.values.astype('object')`. if `lev` was, say, an `Int64Index` then `vals.astype(lev.dtype)` will fail with `ValueError`.

I think `lev._shallow_copy` is more efficient than `Index(....)`.  so i try `_shallow_copy` first.